src/core/qgsvectorlayerlabelprovider.cpp

```diff
@@ -95,6 +95,12 @@ QgsVectorLayerLabelProvider::QgsVectorLayerLabelProvider( const QgsPalLayerSetti
 
 void QgsVectorLayerLabelProvider::init()
 {
+  if ( mLayerGeometryType == QgsWkbTypes::GeometryType::PolygonGeometry &&
```

Comment:
@m-kuhn , are you sure? mLayerGeometryTyp is a QgsWkbTypes::GeometryType enum, which refers to the generic {point,line,polygon,unknown,null} types.